### PR TITLE
Add general background of security and access control

### PR DESCRIPTION
**Pull request by francbartoli**

### CODE REVIEW

File: workshop/content/docs/advanced/security-access-control.md [hunk 1]
```diff
@@ -4,4 +4,26 @@ title: Security and access control
 
 # Security and access control
 
-TODO
+Security is a very huge and complex topic and we do not pretend to cover it in this workshop. It affects the entire development lifecycle and it is recommended to follow security best practices during all the phases like design, development, deployment.
```
Comment (justb4) [inline workshop/content/docs/advanced/security-access-control.md:7]: would rephrase like:
*Security in general is a broad and complex topic, affecting the entire development lifecycle. It is recommended to follow security best practices during all development phases like design, coding and deployment.*

File: workshop/content/docs/advanced/security-access-control.md [hunk 2]
```diff
@@ -4,4 +4,26 @@ title: Security and access control
...
+We want to focus on API security and access control rather than the wide  topic about application security risks.
```
Comment (justb4) [inline workshop/content/docs/advanced/security-access-control.md:8]: rephrase like
*In this workshop we will focus only on API security and access control.*

File: workshop/content/docs/advanced/security-access-control.md [hunk 3]
```diff
@@ -4,4 +4,26 @@ title: Security and access control
...
+!!! Note "Application Security"
+
+    The Open Web Application Security Project (OWASP) [Top Ten document](https://owasp.org/www-project-top-ten/) is a very good tool to ensure the bare minimum against the security risks and manage critical treats that are most likely affecting your code.
+
+API Security is the whole process to protect APIs from attacks. It is part of the more general security guidelines that are being treated in the OWASP Top Ten document. So those recommendations still apply.
+
+Access control is another fundamental part of the Open Web Application Security Project and addresses the Identity and Access Management (IAM) of an API.
+IAM consists of two different parts of a security flow:
+
+- *Authentication* (AuthN) that verifies the user identity. The identity is checked to recognize the user in order to allowing or denying the access
```
Comment (justb4) [inline workshop/content/docs/advanced/security-access-control.md:19]: rephrase:
*- *Authentication* (AuthN) verifies the user's identity in order to allow or deny subsequent access (see next)*

File: workshop/content/docs/advanced/security-access-control.md [hunk 4]
```diff
@@ -4,4 +4,26 @@ title: Security and access control
...
+- *Authorization* (AuthZ) that gives user's permissions to access a resource. The user grants of an identity are checked against the resource's policies to permit the consumption of a API.
```
Comment (justb4) [inline workshop/content/docs/advanced/security-access-control.md:20]: *- *Authorization* (AuthZ)  validates permissions of a user (identity) to access a resource. The permissions of that identity are checked against a resource's policies to (dis)allow access to, for example, (parts of)  an API.*

File: workshop/content/docs/advanced/security-access-control.md [hunk 5]
```diff
@@ -4,4 +4,26 @@ title: Security and access control
...
+
+These parts are usually managed by dedicated infrastructures and solutions which usually provide most of the security requirements out-of-the-box.
+
+!!! Note "OpenAPI Security Specification"
+
+    The OpenAPI specification has very well-defined elements for developers and adopters. The most relevant are:
+        - [Security Scheme Object](https://swagger.io/specification/#security-scheme-object) defines the security schemes that can be used by the operations. Supported schemes are *HTTP Authentication*, an *API Key*, *OAuth2*'s flows and *OpenID Connect*.
+        - [Security Requirement Object](https://swagger.io/specification/#security-requirement-object) defines the list of required security schemes to executean operation.
+
```
Comment (justb4) [inline workshop/content/docs/advanced/security-access-control.md:29]: "executean" --> "execute an"

**Review by justb4 — APPROVED** [review]
Suggested some rephrasing. That is what they are: suggestions, use at own liking.

**Comment by justb4** [comment]
Sorry, now I see this PR is in WIP...should have waited a bit, but was interested in the subject...

**Comment by francbartoli** [comment]
Thanks @justb4 for the suggestions to improve the phrasing!

**Review by justb4 — APPROVED** [review]
Ok, my blessing! Unless others want to review we can merge.
